monai/data/dataset.py

```diff
@@ -289,3 +286,106 @@ def __getitem__(self, index):
+            data = to_tuple(data) + to_tuple(dataset[index])
+        return data
+
+
+class ArrayDataset(ZipDataset):
+    """
+    Dataset for segmentation and classification tasks based on array format input data and transforms.
+    It can apply same random operations for both image transforms and segmentation label transforms.
+    The `transform` can be :py:class:`monai.transforms.Compose` or any other callable object.
+    For example:
+    If train based on Nifti format images without metadata, all transforms can be composed::
+        img_transform = Compose(
+            [
+                LoadNifti(image_only=True),
+                AddChannel(),
+                RandAdjustContrast()
+            ]
+        )
+    If train based on Nifti format images and the metadata, the array transforms can not be composed
+    because several transforms receives multiple parameters or return multiple values. Then Users need
+    to define their own callable method to parse metadata from `LoadNifti` or set `affine` matrix
+    to `Spacing` transform::
+        class TestCompose(Compose):
+            def __call__(self, input_):
+                img, metadata = self.transforms[0](input_)
+                img = self.transforms[1](img)
+                img, _, _ = self.transforms[2](img, metadata["affine"])
+                return self.transforms[3](img), metadata
+        img_transform = TestCompose(
+            [
+                LoadNifti(image_only=False),
+                AddChannel(),
+                Spacing(pixdim=(1.5, 1.5, 3.0)),
+                RandAdjustContrast()
+            ]
+        )
+    Recommend to use dictionary Datasets for complicated data pre-processing.
+    """
+
+    def __init__(
+        self, img_files, img_transform=None, seg_files=None, seg_transform=None, labels=None, label_transform=None
+    ):
+        """
+        Initializes the dataset with the filename lists. The transform `img_transform` is applied
+        to the images and `seg_transform` to the segmentations.
+        Args:
+            img_files (list of str): list of image filenames
+            img_transform (Callable, optional): transform to apply to image arrays
+            seg_files (list of str): if in segmentation task, list of segmentation filenames
+            seg_transform (Callable, optional): transform to apply to segmentation arrays
+            labels (list or array): if in classification task, list of classification labels
+            label_transform (Callable, optional): transform to apply to label arrays
+
+        """
+
+        class Dataset_(Dataset):
+            def set_random_state(self, seed):
+                if self.transform is not None and isinstance(self.transform, Randomizable):
+                    self.transform.set_random_state(seed=seed)
+
+        super().__init__(
+            [Dataset_(img_files, img_transform), Dataset_(seg_files, seg_transform), Dataset_(labels, label_transform)]
```

Comment:
maybe i miss something here, why not use `Dataset` directly, and set the state with
``` 
if isinstance(dataset.transform, Randomizable):
    dataset.transform.set_random_state(seed=seed)
```

Reply:
updated in the latest patch.
Thanks.